spec/models/institute_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Institute, type: :model do
+  subject { institute }
+
+  context 'when all fields contain valid values' do
+    let(:institute) { build(:institute,
+      name: 'Absolutely Awesome Academy') }
+    
+    it { is_expected.to be_valid }
+  end
+
+  context 'when fields have invalid values' do
+    let(:institute) { build(:institute) }
+
+    it 'has an error for a blank name' do
+      subject.name = ''
+
+      subject.valid?
+
+      expect(subject.errors[:name].count).to be > 0
+    end
```

Comment:
If we make the above change, we can just test for assertion that the model is valid

```suggestion
  it { is_expected.to_not be_valid }
```

This does remove the assertion that the `name` field is causing the validation error. If you add in [shoulda matchers](https://github.com/thoughtbot/shoulda-matchers) you could do something like


```ruby
it { is_expected.to validate_presence_of(:name) }
```

Reply:
Yeah, I like that much better - I'll add the gem and refactor to use that instead. It also means I don't need to actually set the name anywhere, since it'll automatically attempt to nil it out. I'll leave the `course_spec` out for now, though it could benefit massively from this.